R/derive_param_tte.R

```diff
@@ -142,10 +159,259 @@
+#'   set to end of study date.
+#' - Then these are passed to derive the time to event parameter with the provided
+#'   parameter descriptions (`PARAMCD` and `PARAM`).
+#' - Note the values of the censor variable (`CNSR`) that were derived here,
+#'   where the first subject has an event and the second does not.
+#' @code
 #' library(tibble)
 #' library(dplyr, warn.conflicts = FALSE)
-#' library(lubridate)
+#' library(lubridate, warn.conflicts = FALSE)
+#'
+#' adsl <- tribble(
+#'   ~USUBJID, ~TRTSDT,           ~EOSDT,            ~NEWDRGDT,
+#'   "01",     ymd("2020-12-06"), ymd("2021-03-06"), NA,
+#'   "02",     ymd("2021-01-16"), ymd("2021-02-03"), ymd("2021-01-03")
+#' ) %>%
+#'   mutate(STUDYID = "AB42")
+#'
+#' ae <- tribble(
+#'   ~USUBJID, ~AESTDT,           ~AESEQ, ~AEDECOD,
+#'   "01",     ymd("2021-01-03"),      1, "Flu",
+#'   "01",     ymd("2021-03-04"),      2, "Cough",
+#'   "01",     ymd("2021-03-05"),      3, "Cough"
+#' ) %>%
+#'   mutate(STUDYID = "AB42")
+#'
+#' ttae <- event_source(
+#'   dataset_name = "ae",
+#'   date = AESTDT,
+#'   set_values_to = exprs(
+#'     EVNTDESC = "AE",
+#'     SRCDOM = "AE",
+#'     SRCVAR = "AESTDTC",
+#'     SRCSEQ = AESEQ
+#'   )
+#' )
+#'
+#' eos <- censor_source(
+#'   dataset_name = "adsl",
+#'   date = EOSDT,
+#'   set_values_to = exprs(
+#'     EVNTDESC = "END OF STUDY",
+#'     SRCDOM = "ADSL",
+#'     SRCVAR = "EOSDT"
+#'   )
+#' )
+#'
+#' derive_param_tte(
+#'   dataset_adsl = adsl,
+#'   event_conditions = list(ttae),
+#'   censor_conditions = list(eos),
+#'   source_datasets = list(adsl = adsl, ae = ae),
+#'   set_values_to = exprs(
+#'     PARAMCD = "TTAE",
+#'     PARAM = "Time to First Adverse Event"
+#'   )
+#' ) %>%
+#'   select(USUBJID, STARTDT, PARAMCD, PARAM, ADT, CNSR, SRCSEQ)
+#'
+#' @caption Adding a by variable (`by_vars`)
+#' @info By variables can be added using the `by_vars` argument, e.g., now for
+#'   each subject the time to first occurrence of each adverse event preferred
+#'   term (`AEDECOD`) should be created as parameters.
+#' @code
+#' derive_param_tte(
+#'   dataset_adsl = adsl,
+#'   by_vars = exprs(AEDECOD),
+#'   event_conditions = list(ttae),
+#'   censor_conditions = list(eos),
+#'   source_datasets = list(adsl = adsl, ae = ae),
+#'   set_values_to = exprs(
+#'     PARAMCD = paste0("TTAE", as.numeric(as.factor(AEDECOD))),
+#'     PARAM = paste("Time to First", AEDECOD, "Adverse Event")
+#'   )
+#' ) %>%
+#'   select(USUBJID, STARTDT, PARAMCD, PARAM, ADT, CNSR, SRCSEQ)
+#'
+#' @caption Handling duplicates (`check_type`)
+#' @info The source records are checked regarding duplicates with respect to the
+#'   by variables and the date and order specified in the source objects.
+#'   By default, a warning is issued if any duplicates are found.
+#'   Note here how the `source_datasets` argument was updated to read in the
+#'   new adverse event source dataset constructed below containing a duplicate
```

Comment:
I don't have a way to re-word this, but I found this a hard to follow???

Reply:
I tried rewording as:

```
#'   Note here how after creating a new adverse event dataset containing a
#'   duplicate date for `"Cough"`, it was then passed to the function using the
#'   `source_datasets` argument - where you see below `ae = ae_dup`.
```

Reply:
love it!